### PR TITLE
Add compatability with modded chairs implemented via TileID.Sets.RoomNeeds.CountsAsChair

### LINKED ISSUE #1: Add compatibility with modded chairs
Mods that add chairs by adding them to the `TileID.Sets.RoomNeeds.CountsAsChair` array (for example, Thorium) aren't included in the `CountChairs()` check as that only checks if the tile type is in the `TileID.Chairs` array, making houses that use those modded chairs invalid once the config option 'requireChairs' is turned on.

### PR DESCRIPTION
**Pull request by CatJet**
Fixes #1

### CODE REVIEW

File: Roommates.cs [hunk 1]
```diff
@@ -48,6 +48,16 @@ private static int CountChairs()
 			for (int j = WorldGen.roomY1; j < WorldGen.roomY2; j++) {
 				if (Main.tile[i, j].TileType == TileID.Chairs) {
 					numChairs++;
+					continue;
+				}
+
```
Comment (NotLe0n) [inline Roommates.cs]: I think this part can be removed, because `CountsAsChair` already has TileID 15 (which is `Chairs`) in the array.
```suggestion
```

**Review by NotLe0n — CHANGES_REQUESTED** [review]

**Comment by CatJet** [comment]
Good Catch!

File: Roommates.cs (hunk 1 above)
Comment (CatJet) [reply Roommates.cs]: Good Catch!

**Comment by NotLe0n** [comment]
Thanks for the PR!

**Pull request merged**